src/history/hs_rec.c

```diff
@@ -909,7 +909,8 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
     for (; ret == 0; ret = hs_cursor->next(hs_cursor)) {
         /* Ignore records that are obsolete. */
         __wt_hs_upd_time_window(hs_cursor, &twp);
-        if (__wt_txn_tw_stop_visible_all(session, twp))
+        if (__wt_txn_tw_stop_visible_all(session, twp) ||
+          (upd_tw != NULL && WT_TIME_WINDOWS_EQUAL(upd_tw, twp)))
```

Comment:
This check is too strict? It is possible that one of the transaction id or timestamp is wiped out because of globally visibility?

Reply:
Yes, the start time window gets cleared when the page is reconciled again. It is better to ignore the start time window if it is globally visible and compare only the stop time window.